test/new-e2e/tests/agent-metric-logs/linux-log-agent/file_tailing_utils_test.go

```diff
@@ -0,0 +1,117 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package logagent
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	fi "github.com/DataDog/datadog-agent/test/fakeintake/client"
+	componentos "github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/stretchr/testify/assert"
+)
+
+// generateLog generates and verifies log contents.
+func generateLog(s *LinuxVMFakeintakeSuite, content string) {
+	// Determine the OS and set the appropriate log path and command.
+	var logPath, cmd, checkCmd string
+	t := s.T()
+
+	// Get OS type using the OSType from vm_client.go from e2e/client package
+	osType := s.Env().VM.GetOSType()
+	var os string
+
+	switch osType {
+	case componentos.WindowsType:
+		os = "windows"
+		t.Log("Generating Windows log.")
+		logPath = "C:\\logs\\hello-world.log"
+		cmd = fmt.Sprintf("echo %s > %s", strings.Repeat(content, 10), logPath)
+		checkCmd = fmt.Sprintf("Get-Content %s", logPath)
+	default: // Assuming Linux if not Windows.
+		os = "linux"
+		t.Log("Generating Linux log.")
+		logPath = "/var/log/hello-world.log"
+		cmd = fmt.Sprintf("echo %s > %s", strings.Repeat(content, 10), logPath)
+		checkCmd = fmt.Sprintf("cat %s", logPath)
+	}
+
+	s.Env().VM.Execute(cmd)
+
+	// Check if the log has been generated.
+	s.EventuallyWithT(func(c *assert.CollectT) {
```

Comment:
Why do we need `EventuallyWithT` here ?
As `VM.Execute(…)` returns the output of the command, I suppose it cannot be asynchronous.